src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -348,6 +348,40 @@ const impl::PtrParticipant ParticipantConfiguration::getParticipant(const std::s
   return *participant;
 }
 
+std::set<std::string> ParticipantConfiguration::knownParticipants() const
+{
+  std::set<std::string> names;
+  for (const auto &p : _participants) {
+    names.insert(p->getName());
+  }
```

Comment:
If we want to avoid more of these for loops, not sure. We should provide maybe a small config test here.
```suggestion
    std::transform(_participants.begin(), _participants.end(), std::inserter(names, names.end()),
        [](const auto& p) { return p->getName(); });
```